menuinst/_schema.py

```diff
@@ -340,8 +351,17 @@ class Platforms(BaseModel):
 class MenuItem(BaseModel):
     "Instructions to create a menu item across operating systems."
 
-    name: constr(min_length=1) = ...
-    "The name of the menu item."
+    name: Union[
+        constr(min_length=1),
+        Dict[constr(regex=r"^target_environment_is_(not_)?base$"), constr(min_length=1)],
```

Comment:
Same here. We can even create a TypeVar at the top-level if needed.

Reply:
I ended up using a `TypedDict`. Using `Literal` as dictionary keys did not get processed well by the schema script (it discarded the key restrictions)

Reply:
`Literal` does not seem to produce the outcome that's expected because the `schema_json` function drops the `Literal` entirely.

I tried a `TypedDict`, but that will make things more complicated since `typing_extensions` are needed for python 3.8 and the schema fails to validate (even though it looks okay to me).

Reply:
I think I got it to work now. The way it is now should make it extensible in case we want to add additional naming conditions later.